src/Sparrow/Json/BlittableJsonReaderObject.cs

```diff
@@ -483,5 +485,180 @@ public void CopyTo(byte* ptr)
             Memory.Copy(ptr, _mem, _size);
         }
 
+        public void BlittableValidation(int size)
+        {
+            byte offset;
+            var currentSize = size - 1;
+            int rootPropOffsetSize;
+            int rootPropIdSize;
+
+            var rootToken = TokenValidation(*(_mem + currentSize), out rootPropOffsetSize, out rootPropIdSize);
+            if (rootToken != BlittableJsonToken.StartObject)
+                throw new InvalidDataException("Illegal root object");
+            currentSize--;
+
+            var propsOffsetList = ReadVariableSizeIntInReverse(_mem, currentSize, out offset);
+            currentSize -= offset;
+
+            var rootMetadataOffset = ReadVariableSizeIntInReverse(_mem, currentSize, out offset);
+            currentSize -= offset;
+
+            if ((propsOffsetList > currentSize) || (propsOffsetList <= 0))
+                throw new InvalidDataException("Properties names offset not valid");
+
+            int propNamesOffsetSize;
+            var token = (BlittableJsonToken)(*(_mem + propsOffsetList));
+            propNamesOffsetSize = ProcessTokenOffsetFlags(token);
+            if (((token & (BlittableJsonToken)0xC0) != 0) || ((TypesMask & token) != 0x00))
```

Comment:
token can only have offset size , so I am checking that the property id size is 0